sdcm/cluster.py

```diff
@@ -2193,6 +2193,27 @@ def stop_scylla_server(self, verify_up=False, verify_down=True, timeout=300, ign
         if verify_down:
             self.wait_db_down(timeout=timeout)
 
+    @retrying(n=2, sleep_time=5, allowed_exceptions=(CommandTimedOut, ),
+              message="Failed to stop scylla.server, retrying...", raise_on_exceeded=False)
+    def safe_stop_scylla_server(self, verify_up=False, verify_down=True, timeout=300, ignore_status=True):
+        """Stop scylla in safe mode
+
+        Try to stop scylla-server and don't interrupt test and don't raise exception
+        UnexpectedExit, Failure and CommandTimedOut if "scylla-server stop" failed
+        or scylla-server was stopped gracefully
+
+        :param verify_up: verify scylla-server is up before stopping, defaults to False
+        :type verify_up: bool, optional
+        :param verify_down: verify scylla-server was stopped, defaults to True
+        :type verify_down: bool, optional
+        :param timeout: command execution time, defaults to 300
+        :type timeout: number, optional
+        :param ignore_status: ignore command exit status, defaults to False
+        :type ignore_status: bool, optional
+        """
```

Comment:
I don't think it's needed, wrapping the whole thing just to pass "ignore status" and add a retry is overkill.

Probably the correct thing to do is to anayze all our calls to "stop_scylla_server" and check if "retry" is dangerous there or would give  something that isn't expected to the user...

@bentsi / @fruch WDYT?

Reply:
We already have retry on "stop_scylla_server"!
Just add your exception and that's it.

I reviewed all places, it looks safe

Reply:
If add to allowed exception to stop_scylla_server it will affect all places. And we may missed some scylla issue.
In such implementation i think it will be safer at least we will get goal: workaround only for Drainer

Reply:
If set ignore status only, then exception CommandTimedOut will be raised and will stop the test

Reply:
We should have very solid reasons to add a new flavor of the method that starts scylla-server.
In this case, there is no one:
- if the issue is global (happens on every stop) we should add CommandTimedOut to the `allowed_exceptions` on `sdcm.cluster.BaseNode.stop_scylla_server`
- if the issue happens only during drain nemesis the solution should be in the code of the drain nemesis